numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1724,6 +1741,10 @@ mapiter_@name@(PyArrayMapIterObject *mit)
                         NPY_AUXDATA_FREE(transferdata);
                         return -1;
                     }
+                    if(subspace_ptrs[0] == self_ptr &&
+                       subspace_ptrs[1] == mit->extra_op_ptrs[0]) {
+                        skip <<= 1;
```

Comment:
I think you are right. I changed it to 1 and all tests passes too. The offset from selfptr and the first iteration is always a constant, and we can calculate it on the first iteration.